### PR TITLE
Added in-memory configuration for app settings in the WebApplicationFactory

### PR DESCRIPTION
**Pull request by SaiSDET**
- Made email address configurable from the TestWebApplicationFactory class

### CODE REVIEW

**Comment by SaiSDET** [comment]
@dotnet-policy-service agree [company="NA"]

**Comment by SaiSDET** [comment]
@dotnet-policy-service agree

File: fundamentals/minimal-apis/samples/MinApiTestsSample/IntegrationTests/Helpers/TestWebApplicationFactory.cs [hunk 1]
```diff
@@ -11,6 +12,11 @@ public class TestWebApplicationFactory<TProgram>
 {
     protected override IHost CreateHost(IHostBuilder builder)
     {
+        builder.ConfigureHostConfiguration(config =>
+        {
+            config.AddInMemoryCollection(new Dictionary<string, string?> { { "EmailAddress", "testhello@microsoft.com" } });
```
Comment (Rick-Anderson) [inline fundamentals/minimal-apis/samples/MinApiTestsSample/IntegrationTests/Helpers/TestWebApplicationFactory.cs:17]:
```suggestion
            config.AddInMemoryCollection(new Dictionary<string, string?> { { "EmailAddress", "test1@Contoso.com" } });
```

File: fundamentals/minimal-apis/samples/MinApiTestsSample/WebMinRouteGroup/appsettings.json [hunk 2]
```diff
@@ -5,5 +5,6 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "EmailAddress": "hello@microsoft.com"
```
Comment (Rick-Anderson) [inline fundamentals/minimal-apis/samples/MinApiTestsSample/WebMinRouteGroup/appsettings.json:9]:
```suggestion
  "EmailAddress": "test1@Contoso.com"
```

**Comment by Rick-Anderson** [comment]
@fiyazbinhasan please review

**Comment by fiyazbinhasan** [comment]
@Rick-Anderson @SaiSDET looks good! @SaiSDET Would be great if you refer to the actual issue in the PR, e.g. Fixes [issue_number] ✌️

**Pull request merged**